notebooks/dc2/catalog_generation/matchingLensGalaxies.ipynb

```diff
@@ -0,0 +1,6293 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.hist(np.log10(sigma_fp))\n",
+    "plt.xlabel('Log(Velocity Dispersion)')\n",
+    "plt.ylabel('DDF galaxies')\n",
+    "plt.title('Velocity Dispersion calculated from Fundamental Plane')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Match to OM10"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 89,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Load OM10\n",
+    "hdulist = fits.open('../../../data/om10_qso_mock.fits')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 90,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "om10_data = hdulist[1].data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 91,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Only keep om10 systems where the host galaxy is within the redshift range of cosmoDC2\n",
+    "om10_data = om10_data[np.where(om10_data['ZSRC'] <= 3.0)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 92,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def match_to_lenscat(vel_disp, redshift, om10_array, density=1.0, seed=None):\n",
+    "    \n",
+    "    \"\"\"\n",
+    "    Match DC2 galaxies to lens galaxies in OM10 based upon velocity dispersion and redshift.\n",
+    "    We set the threshold for a match to be 0.05 in dex in each parameter.\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    if seed is None:\n",
+    "        rand_state = np.random.RandomState()\n",
+    "    else:\n",
+    "        rand_state = np.random.RandomState(seed)\n",
```

Comment:
Won't the `seed` option give you the same random number for all the rows? I could be wrong...